## **Penguins species extraction**

In this notebook we take a dataset that contains culmen length and depth, flipper length, bmi and gender of a penguin and the task is to exctrat different sub-species od penguins based on given characteristics. In the approach we will focus on unsupervised ML methods as well as technology of dimensionality reduction.

## Impotrs

In [1]:
import pandas as pd

## Load data

In [2]:
df = pd.read_csv('penguins.csv')
df

culmen_length_mm  culmen_depth_mm  flipper_length_mm  body_mass_g  gender
0                39.1             18.7              181.0       3750.0    MALE
1                 NaN             17.4              186.0       3800.0  FEMALE
2                40.3             18.0              195.0       3250.0  FEMALE
3                36.7             19.3              193.0       3450.0  FEMALE
4                39.3             20.6              190.0       3650.0    MALE
..                ...              ...                ...          ...     ...
327              47.2             13.7              214.0       4925.0  FEMALE
328               NaN              NaN              215.0       4850.0  FEMALE
329              50.4             15.7              222.0       5750.0    MALE
330              45.2             14.8              212.0       5200.0  FEMALE
331              49.9             16.1              213.0       5400.0    MALE

[332 rows x 5 columns]

## **EDA (Exploratory data analysis)**

## Getting insights about the dataset

In [3]:
df.shape

(332, 5)

This means that this dataset has 332 rows and 5 columns.

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 332 entries, 0 to 331
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   culmen_length_mm   328 non-null    float64
 1   culmen_depth_mm    329 non-null    float64
 2   flipper_length_mm  327 non-null    float64
 3   body_mass_g        332 non-null    float64
 4   gender             331 non-null    object 
dtypes: float64(4), object(1)
memory usage: 13.1+ KB


Here we can see the columns and their data types. If we look closer to the Not-Null Count and amount of entries we have we can see that there is a little amount of missing values in everyt column except bmi

In [8]:
data_types = df.dtypes
column_names = df.columns
num_columns = []
for column in column_names:
    if data_types[column] == 'object':
        print(f"Column name is: {column}")
        print('This column contains categorical data')
        print(f"And the uniques values are:\n{df[column].unique()}\n")
    else:
        num_columns.append(column)
print(f'This columns contain numerical data:\n{num_columns}')

Column name is: gender
This column contains categorical data
And the uniques values are:
['MALE' 'FEMALE' nan]

This columns contain numerical data:
['culmen_length_mm', 'culmen_depth_mm', 'flipper_length_mm', 'body_mass_g']


As we can see the only gender column is the one that we will have to encode. It is shown again that there is a missing value in the gender column.

## **Clustering algorithms experiments**

## Before scaling

## After scaling

## **What can be improved**
- pass